clients/include/testing_gbmv_strided_batched.hpp

```diff
@@ -0,0 +1,512 @@
+                              rocblas_status_invalid_size);
+
+        return;
+    }
+
+    //quick return
+    if(!M || !N || !batch_count)
+    {
+        static const size_t safe_size = 100; // arbitrarily set to 100
+        device_vector<T>    dA1(safe_size);
+        device_vector<T>    dx1(safe_size);
+        device_vector<T>    dy1(safe_size);
+        if(!dA1 || !dx1 || !dy1)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        EXPECT_ROCBLAS_STATUS(rocblas_gbmv_strided_batched<T>(handle,
+                                                              transA,
+                                                              M,
+                                                              N,
+                                                              KL,
+                                                              KU,
+                                                              &h_alpha,
+                                                              dA1,
+                                                              lda,
+                                                              stride_A,
+                                                              dx1,
+                                                              incx,
+                                                              stride_x,
+                                                              &h_beta,
+                                                              dy1,
+                                                              incy,
+                                                              stride_y,
+                                                              batch_count),
+                              rocblas_status_success);
+
+        return;
+    }
+
+    size_A = size_A + static_cast<size_t>(stride_A) * static_cast<size_t>(batch_count - 1);
+    size_x = size_x + static_cast<size_t>(stride_x) * static_cast<size_t>(batch_count - 1);
+    size_y = size_y + static_cast<size_t>(stride_y) * static_cast<size_t>(batch_count - 1);
+
+    // Naming: dK is in GPU (device) memory. hK is in CPU (host) memory
+    host_vector<T> hA(size_A);
+    host_vector<T> hx(size_x);
+    host_vector<T> hy_1(size_y);
+    host_vector<T> hy_2(size_y);
+    host_vector<T> hy_gold(size_y);
+
+    device_vector<T> dA(size_A);
+    device_vector<T> dx(size_x);
+    device_vector<T> dy_1(size_y);
+    device_vector<T> dy_2(size_y);
+    device_vector<T> d_alpha(1);
+    device_vector<T> d_beta(1);
+    if((!dA && size_A) || (!dx && size_x) || ((!dy_1 || !dy_2) && size_y) || !d_alpha || !d_beta)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    // Initial Data on CPU
+    rocblas_seedrand();
+    // Init a lda * N matrix, not M * N
+    rocblas_init<T>(hA, lda, N, lda, stride_A, batch_count);
+    rocblas_init<T>(hx, 1, dim_x, abs_incx, stride_x, batch_count);
+
+    if(rocblas_isnan(arg.beta))
```

Comment:
This is seen in some other functions, eg. gemv.
If arg.beta is NaN it is interpreted as 0 in get_beta(). I believe the reason we have this here is to test that when beta = 0, y can be set to NaN and not affect anything since we don't have to do the multiplication.

Reply:
That is some pretty strange logic so beta=Nan actually tests beta=0 and y=Nan , not blaming you of course.

Reply:
> > This is seen in some other functions, eg. gemv.
> > If arg.beta is NaN it is interpreted as 0 in get_beta(). I believe the reason we have this here is to test that when beta = 0, y can be set to NaN and not affect anything since we don't have to do the multiplication.

> That is some pretty strange logic so beta=Nan actually tests beta=0 and y=Nan , not blaming you of course.

It was the most expedient solution, and it kind of "makes sense" in a self-documenting way, because when you set `beta = NaN` you're essentially saying "Make beta 0, and all other problem parameters being equal, you can initialize C with `NaN`s and not expect `NaN`s in the output".

This was simpler than creating separate test functions to specifically test `NaN` and duplicating the rest of the test code, and it was simpler than adding an extra flag to indicate `NaN` testing to the test `Arguments`, which is already very bloated. Hey, what if we simply set `beta` to `NaN` to indicate we should do `NaN` tests? Then the test sets `beta` to `0`, and `C` to `NaN`, and we expect no `NaN` in the output, and there's almost no other code changes needed. We're not losing any other tests, because how often do you have to test `beta` with an actual `NaN` value?